tls/s2n_record_write.c

```diff
@@ -90,6 +93,7 @@ S2N_RESULT s2n_record_max_write_payload_size(struct s2n_connection *conn, uint16
         if (send_buffer_override < max_record_size) {
             size_t overhead = (max_record_size - *max_fragment_size);
             RESULT_ENSURE_GT(send_buffer_override, overhead);
+            RESULT_ENSURE((send_buffer_override - overhead) >= 2, S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
             *max_fragment_size = send_buffer_override - overhead;
```

Comment:
Why not check *max_fragment_size before this method returns? You could also define a new S2N_TLS_MINIMUM_FRAGMENT_LENGTH constant for use everywhere you're currently hardcoding "2". That would save you a lot of comments about alerts :)

Reply:
Yeah. This makes sense. I added the two different lines based on two different test that failed. It's kinda nice to know exactly what went wrong based on the line number where we bail. On the other hand that kind of checking is not easily adaptable to future changes.

On the `S2N_TLS_MINIMUM_FRAGMENT_LENGTH` I don't like the name. It implies that messages can't be fragmented any smaller. Really what we want is the max of the minimum fragment lengths across handshake, application data and alert messages. MAXIMUM_OF_MINIMUMS is overly verbose and potentially confusing. I also didn't want to use MINIMUM ALERT FRAGMENT LENGTH, cause I thought it was odd to reference the alert messages so directly. 

Not being able to come up with a good name is part of why I didn't add exactly this variable.

This comment caused me to revisit the name and I think I came up with a better option: `S2N_TLS_MINIMUM_FRAGMENT_RESERVE_LENGTH`. This name accurately expresses the idea that this is not the minimum fragment length but rather the minimum space we need to reserve, in the same sense of a Rust Vec.